libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/dump/MapboxDumpHandler.kt

```diff
@@ -0,0 +1,77 @@
+package com.mapbox.navigation.core.internal.dump
+
+import java.io.FileDescriptor
+import java.io.PrintWriter
+
+/**
+ * This will process commands from the dumpsys. The expected format is
+ */
+internal class MapboxDumpHandler {
+
+    fun handle(fd: FileDescriptor, writer: PrintWriter, args: Array<String>?) {
+        val handled = handleArguments(fd, writer, args) != null
+
+        writer.println(args?.joinToString())
+        // Show help command when no arguments were handled.
+        if (!handled) {
+            handleArguments(fd, writer, arrayOf(HelpDumpInterceptor().command().command))
+        } else {
+            writer.println("Finished handling the commands: ${args?.toList()}")
+        }
+    }
+
+    private fun handleArguments(
+        fd: FileDescriptor,
+        writer: PrintWriter,
+        args: Array<String>?
+    ): Unit? {
+        val matches: List<Pair<MapboxDumpInterceptor?, List<String>>> = args
+            ?.map { command -> Pair(command, MapboxDumpRegistry.getInterceptor(command)) }
+            ?.groupBy { it.second }?.map { groupedEntry ->
+                groupedEntry.key to groupedEntry.value.map { it.first }
+            } ?: return null
+
+        var handled = false
+        matches.forEach { match ->
+            val interceptor = match.first
+            val commands = match.second
+            if (interceptor == null) {
+                writer.println("Unrecognized commands: ${commands.joinToString()}")
+            } else {
+                writer.println("Processing commands: ${commands.joinToString()}")
+                interceptor.intercept(fd, writer, commands.toMapboxDumpCommand())
+                handled = true
+            }
+        }
+        return if (handled) Unit else null
```

Comment:
It was just an opinion when I was building it.

The thought was, it's user friendly to print the help command when no arguments are passed.

I would want the caller (NavigationNotificationService) to handle the result if we have a plan to make other services use this framework. Maybe we should, but that would also be a super simple change so figured we can worry about it when/if it happens

Reply:
Yeah, I see that, I mean why would you want to check null/non-null? Why not check a boolean instead? I think it's common to return true if something was handled/processed/whatever, false otherwise.

Reply:
OK I see what you mean. This logic was confusing so I made another pass to clean it up.

Now this function returns the list of interceptors that recognized the command. That way it is empty if nothing recognizes the command.